README.md

```diff
@@ -1,62 +1,150 @@
+   ```
+
+   Pyramid issue: https://github.com/Pylons/pyramid/issues/434
+
+   `pyramid_sanity` catches this and turns it into a 400 Bad Request response.
+
+3. A bad form submission can crash the app when the app tries to access `request.POST`.
+
+   With the example app above this will raise an uncaught `ValueError` from WebOb:
+
+   ```terminal
+   curl --request POST --url http://localhost:6543/foo --header 'content-type: multipart/form-data'
+   ```
+
+   Pyramid issue: https://github.com/Pylons/pyramid/issues/1258
 
-* This plugin provides a way to catch a number of encoding errors which happen
- which are otherwise difficult to handle in the Pyramid framework
-* Various crashes should now become `BadRequest` causing 400 errors:
-   * Calling with bad unicode query parameters
-   * Calling with bad unicode in the URL
-   * Calling with poorly specified form boundaries
-* In each case a specific exception is raised so you can handle it how you would
-like to
-* There is also one fix:
-   * Pyramid emitting redirects with unicode in the URL
+   `pyramid_sanity` catches this and turns it into a 400 Bad Request response.
+
+4. If the app returns a redirect with a badly encoded location it can crash the WSGI server.
+
+   With the example app above this will raise an uncaught `AttributeError` from `wsgiref.simple_server`:
+
+   ```terminal
+   curl http://localhost:6543/redirect
+   ```
+
+   `pyramid_sanity` intercepts this and safely encodes the redirect location.
 
 Usage
 -----
 
 ```python
-config.add_settings({
-    "pyramid_sanity.ascii_safe_redirects": False
-    # See below for all available settings
-})
-
-# Important! Add as near to the end of your config as possible:
-config.include("pyramid_sanity")
+with Configurator() as config:
+    # Add this as near to the end of your config as possible:
+    config.include("pyramid_sanity")
 ```
 
 Settings
 --------
 
-| Pyramid setting | Default | Effect |
-|-----------------|--------|---------|
-| `pyramid_sanity.disable_all` | `False` | Disable all features, so they can be selectively enabled
-| `pyramid_sanity.check_form` | `True` | Check for badly declared forms
-| `pyramid_sanity.check_params` | `True` | Check for encoding errors in URL parameters
-| `pyramid_sanity.check_path` | `True` | Check for encoding errors in the URL path
-| `pyramid_sanity.ascii_safe_redirects` | `True` | Ensure redirects do not include raw unicode characters
+By default all fixes are enabled. You can disable them individually with settings:
+
+```python
+with Configurator() as config:
+    config.add_settings({
+        # Don't check for badly declared forms.
+        "pyramid_sanity.check_form": False,
+
+        # Don't check for badly encoded query params.
+        "pyramid_sanity.check_params": False,
+
+        # Don't check for badly encoded URL paths.
+        "pyramid_sanity.check_path": False,
+
+        # Don't safely encode redirect locations.
+        "pyramid_sanity.ascii_safe_redirects": False
```

Comment:
Moved the docs for the settings out of a table and into a handy copy-pastable code snippet